uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/calendar-instance_test.js

```diff
@@ -9,75 +9,52 @@ define([
   'use strict';
 
   describe('CalendarInstance', function () {
-    var CalendarInstance;
+    var instance, mockedCalendar;
 
     beforeEach(module('leave-absences.models.instances'));
-
-    beforeEach(inject([
-      'CalendarInstance',
-      function (_CalendarInstance_) {
-        CalendarInstance = _CalendarInstance_.init(mockData.daysData().values[0]);
-      }]
-    ));
+    beforeEach(inject(['CalendarInstance', function (CalendarInstance) {
+      mockedCalendar = mockData.daysData().values[0];
+      instance = CalendarInstance.init(mockedCalendar);
+    }]));
 
     describe('init()', function () {
       var key, date;
 
       beforeEach(function () {
-        // check with first key of the object
-        key = Object.keys(CalendarInstance.days)[0];
-        date = moment(CalendarInstance.days[key].date).valueOf().toString();
+        key = Object.keys(instance.days)[0];
+        date = moment(instance.days[key].date).valueOf();
       });
 
-      it('dates arrays has been converted to an object with proper timestamp values', function () {
-        expect(key).toBe(date);
+      it('converts the dates array to an object with timestamps as keys', function () {
+        expect(+key).toBe(date);
```

Comment:
the `key` was a string, the `date` an integer, so it was necessary to call `toString()` on the `date`. I simply cast the `key` to be integer, less code to read now